stdlib/public/Darwin/Foundation/Data.swift

```diff
@@ -2084,9 +2084,8 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
                 // ... and append the rest byte-wise, buffering through an InlineData.
                 var buffer = InlineData()
                 while let element = iter.next() {
-                    if buffer.count < buffer.capacity {
-                        buffer.append(byte: element)
-                    } else {
+                    buffer.append(byte: element)
```

Comment:
To confirm my understanding:
- `buffer` has a fixed capacity
- if it's at capacity after appending one element, you append the buffer to `_representation` and reset the buffer
- the bug was, you were detecting one less-than capacity and not appending under those circumstances, so that element was dropped
- you're sure it's not possible to ever append when already at capacity

Right?

Reply:
Correct on all accounts. The capacity is fixed to what we can stack-allocate and once we hit that capacity, we append what we had on the stack. The bug is that we threw away `element` previously in the case where we appended `buffer`; we now append it unconditionally with the knowledge that there is always space in the buffer before doing so.